src/board_builder/board_builder.py

```diff
@@ -0,0 +1,261 @@
+                corners_list: list[list[float]] = []
+                for corner in marker_snapshot.corner_image_points:
+                    corners_list.append([corner.x_px, corner.y_px])
+                marker_corners = MarkerCorners(
+                    detector_label=detector_name,
+                    marker_id=int(marker_snapshot.label),
+                    points=corners_list,
+                    timestamp=timestamp
+                )
+                self.pose_solver.add_marker_corners([marker_corners])
+
+        new_detector_poses = self.pose_solver.get_detector_poses()
+        for pose in new_detector_poses:
+            if pose.target_id not in self._detector_poses_median:
+                self._detector_poses_median[pose.target_id] = PoseLocation(pose.target_id)
+            self._detector_poses_median[pose.target_id].add_matrix(np.array(pose.object_to_reference_matrix.values).reshape(4, 4), str(timestamp))
+        for label in self._detector_poses_median:
+            pose = Pose(
+                target_id=label,
+                object_to_reference_matrix=self._detector_poses_median[label].get_median_pose().object_to_reference_matrix,
+                solver_timestamp_utc_iso8601=str(timestamp)
+            )
+            self.detector_poses.append(pose)
+        self.pose_solver.set_detector_poses(self.detector_poses)
+
+    def collect_data(self, detector_data: dict[str, list[MarkerSnapshot]]):
+        """ Collects data of relative position and is entered in matrix. Returns a dictionary of its corners"""
+        detector_data = self._filter_markers_appearing_in_multiple_detectors(detector_data)
+        if all(isinstance(v, list) and len(v) == 0 for v in detector_data.values()):
+            return
+        timestamp = datetime.datetime.utcnow()
+        corners_dict = {}
+        self.target_poses = []
+        self._solve_pose(detector_data, timestamp)
+        for index, pose in enumerate(self.target_poses):
+            # R R R T
+            # R R R T
+            # R R R T
+            # 0 0 0 1
+
+            pose_values = pose.object_to_reference_matrix.values
+            pose_matrix = np.array(pose_values).reshape(4, 4)
+
+            for other_pose in self.target_poses:
+                if other_pose != pose:
+                    other_matrix_values = other_pose.object_to_reference_matrix.values
+                    other_pose_matrix = np.array(other_matrix_values).reshape(4, 4)
+                    relative_transform = self._calculate_relative_transform(pose_matrix, other_pose_matrix)
+                    matrix_index = self._find_matrix_input_index(pose.target_id, other_pose.target_id)
+
+                    if not self._relative_pose_matrix[matrix_index[0]][matrix_index[1]]:
+                        new_pose_location = PoseLocation(pose.target_id)
+                        new_pose_location.add_matrix(relative_transform, str(datetime.datetime.now()))
```

Comment:
`datetime.datetime.now()` is local time. We should not use local time for any time computations because the code would break whenever switching to/from daylight saving. If the timestamp is informational only (e.g., used for log messages) then local time is fine. For computations, `datetime.datetime.utcnow()` should be used instead.

 `str(datetime.datetime.now())` returns `'2024-07-29 21:39:28.218641'` this is not ISO 8601. We can use `datetime.datetime.now().isoformat()`, which returns `'2024-07-29T21:41:18.201554'` to get time in a format that is universally recognized.

It would be better to store the datetime object (`datetime.datetime.now()`) in memory to avoid all the expensive conversions whenever we want to do any computations. The datetime object can be converted to UTC or local time string for display, logging, etc.